x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java

```diff
@@ -128,9 +128,8 @@ public Void run() {
     //private final Environment env;
     protected final Licensing licensing;
     // These should not be directly accessed as they cannot be overridden in tests. Please use the getters so they can be overridden.
-    private static final SetOnce<XPackLicenseState> licenseState = new SetOnce<>();
-    private static final SetOnce<SSLService> sslService = new SetOnce<>();
-    private static final SetOnce<LicenseService> licenseService = new SetOnce<>();
+
+    private final List<XPackSharedPlugin> sharedPlugins = new ArrayList<>();
```

Comment:
Naming-wise we could rename `XPackPlugin` to `XPack` and then let `XPackSharedPlugin` be called `XPackPlugin`, but deferred this to a follow-up.